### ISSUE TITLE
Why? What? Who? How?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/phoenix-uk-postcode-finder-example#1]
# Why?

"Store locators" are a common use-case for most retailers or organisations with physical premises.
We have built them for a few of our clients and it's time to make a _generic_ example that _anyone_ can understand and learn from! 

# What? 

A tutorial showing how to build a "find my nearest `X`" (_AKA "store locator"_) using freely available location data, PostgreSQL, ETS and Elixir/Phoenix. 

> Note: The end result should be a reasonably attractive browser application that allows the person to input a UK postcode and will display both their postcode location and the location of the `n` nearest `X` on a map with the list below the map showing the distance to the `X`. 
Clicking on the link in the _list_ should open the default map application on the device or if it's a desktop/laptop should open Google Maps in full-screen using a `target="_blank"`. 

# Who?

People who want to learn about Geolocation, Google Maps, Distance calculation and rapid lookups in medium-sized datasets. 
Anyone who wants to learn how to build a real-world "store finder" in Elixir/Phoenix.

# _How_?

@RobStallion please _estimate_ how long it will take you to convert the knowledge you gained while building the feature for our client: https://github.com/club-soda/club-soda-guide/pull/291 into a stand-alone example app that allows people to input their postcode and find the _nearest_ `X` (_where `X` is your choice of "vegan restaurant", "train station", "free outdoor gym" ... - or whatever data is publicly available, interesting to you and easy to parse..._ 😉)

**Comment by RobStallion** [comment]
@nelsonic do you know any good tools for getting a json of address info for a random group of things, e.g. vegan restaurants, outdoor gyms? 

I don't _**need**_ it for the example as I can just use any random addresses but I thought that it might be a nice touch, and from your comment so did you...
> (where X is your choice of "vegan restaurant", "train station", "free outdoor gym" ... - or whatever data is publicly available, interesting to you and easy to parse... 😉)

I haven't yet done much googling around for a tool yet. I thought I would ask beforehand while I worked on other parts of the example.

**Comment by nelsonic** [comment]
@robstallion yes, but it depends on what data you're going with.

**Comment by RobStallion** [comment]
@nelsonic Outdoor gyms?

**Comment by nelsonic** [comment]
@robstallion agreed. (_what did Google say when you asked for a "list of UK outdoor gyms"...?_) 🤔

**Comment by RobStallion** [comment]
https://data.gov.uk/dataset/bbd56d2d-0899-4be2-bb24-f02ff50c2592/outdoor-sports-facilities/datafile/a6cb4fcc-6d13-4f28-a62e-40fc230c1396/preview

![image](https://user-images.githubusercontent.com/15571853/54018881-b385fc80-4181-11e9-88ef-8694385f30d8.png)

**Comment by nelsonic** [comment]
@RobStallion The data you have found and used is only for Leeds Council. That's a **_good_ start**. 👍 

I've reached out to Georgie & Matt Delaney co-founders of The Great Outdoor Gym Company (TGO)
![image](https://user-images.githubusercontent.com/194400/54042068-2ebad300-41c1-11e9-9d8d-a5cc6b3902c1.png)
You know, the company that makes the outdoor Gym Equipment you see all over London (and the UK!) 😉 
![image](https://user-images.githubusercontent.com/194400/54042047-21054d80-41c1-11e9-803b-58811fdbbaba.png)
to see if they can share their list of Outdoor Gym locations. (I CC'd you in the email so you are aware)
But you should move forward with the Leeds data in case they don't get back to us fast enough.

**Comment by nelsonic** [comment]
An alternative source of data might be https://calisthenics-parks.com/maps
![image](https://user-images.githubusercontent.com/194400/54065992-38275800-4220-11e9-9e77-6d62c5054221.png)
